DigitalLearningSolutions.Web/Views/LearningPortal/Shared/_Dates.cshtml

```diff
@@ -18,50 +18,45 @@
+  <div class="nhsuk-grid-row nhsuk-grid-column-one-half">
+    <div class="nhsuk-grid-column-three-quarters">
+      <span class="nhsuk-label--s nhsuk-u-margin-bottom-2" id="@Model.Id-complete-by-title" aria-describedby="@Model.Id-name">Complete by</span>
+      <hr class="nhsuk-u-margin-top-0 nhsuk-u-margin-bottom-2" aria-hidden="true" />
     </div>
-  }
-  else if (currentLearningItemViewModel is SelfAssessmentCardViewModel)
-  {
     <div class="nhsuk-grid-column-one-half nhsuk-u-one-half">
-      <a class="nhsuk-button nhsuk-button--secondary small-edit-button" aria-describedby="@Model.Id-name" aria-label="Edit complete by date" asp-route-selfAssessmentId="@Model.Id" asp-action="SetSelfAssessmentCompleteByDate">
-        Edit
-      </a>
-    </div>
-  }
-  else if (currentLearningItemViewModel is LearningResourceCardViewModel)
-  {
-    <div class="nhsuk-grid-column-one-half nhsuk-u-one-half">
-      <a class="nhsuk-button nhsuk-button--secondary small-edit-button"
-         aria-describedby="@Model.Id-name"
-         aria-label="Edit complete by date">
-        Edit
-      </a>
+      <p class="@currentLearningItemViewModel.DateStyle()" aria-describedby="@Model.Id-name @Model.Id-complete-by-title" name="complete-by-date">
+        @(currentLearningItemViewModel.CompleteByDate == null ?
+          Html.Raw("<span class='nhsuk-u-visually-hidden'>No complete by date set </span>-") :
+          Html.Raw($"<span class='nhsuk-u-visually-hidden'>{currentLearningItemViewModel.DueByDescription()}</span>{currentLearningItemViewModel.CompleteByDate?.ToShortDateString()}"))
+      </p>
     </div>
-  }
-</div>
+    @if (currentLearningItemViewModel is CurrentCourseViewModel course && (course.CompleteByDate == null || course.SelfEnrolled)) {
+      <div class="nhsuk-grid-column-one-half nhsuk-u-one-half">
+        <a class="nhsuk-button nhsuk-button--secondary small-edit-button" aria-describedby="@Model.Id-name" aria-label="Edit complete by date" asp-action="SetCurrentCourseCompleteByDate" asp-route-id="@Model.Id">
+          Edit
+        </a>
+      </div>
+    } else if (currentLearningItemViewModel is SelfAssessmentCardViewModel) {
+      <div class="nhsuk-grid-column-one-half nhsuk-u-one-half">
+        <a class="nhsuk-button nhsuk-button--secondary small-edit-button" aria-describedby="@Model.Id-name" aria-label="Edit complete by date" asp-route-selfAssessmentId="@Model.Id" asp-action="SetSelfAssessmentCompleteByDate">
+          Edit
+        </a>
+      </div>
+    } else if (currentLearningItemViewModel is LearningResourceCardViewModel) {
+      <div class="nhsuk-grid-column-one-half nhsuk-u-one-half">
+        <a class="nhsuk-button nhsuk-button--secondary small-edit-button"
+           aria-describedby="@Model.Id-name"
+           aria-label="Edit complete by date"
+           asp-action="SetCurrentActionPlanItemCompleteByDate"
+           asp-route-learningLogItemId="@Model.Id">
+          Edit
+        </a>

```

Comment:
Just want to double check, as there are a number of slight formatting changes on this page: is the setting of this action is the only actual change on this page?

Reply:
Yep that's the only change I made!